src/Enzo/enzo_EnzoMethodPmDeposit.cpp

```diff
@@ -132,29 +165,22 @@ void EnzoMethodPmDeposit::compute ( Block * block) throw()
                ((be == 4) ? "single" : ((be == 8) ? "double" : "quadruple")),
                (ba == be));
 
-      // Accumulate particle density using CIC
 
-      enzo_float cosmo_a=1.0;
-      enzo_float cosmo_dadt=0.0;
-      EnzoPhysicsCosmology * cosmology = enzo::cosmology();
-    
-      if (cosmology) {
+      enzo_float dens = 0.0;
+      int ia_m = -1;
+
+      // Determine if particle has a constant mass, or if mass varies
+      // and we need to use the mass attribute
+      if (particle.has_constant (it,"mass")){
 
-        double time = block->time();
-        double dt   = block->dt();
-        cosmology->compute_expansion_factor (&cosmo_a,&cosmo_dadt,time+alpha_*dt);
-      
+        const int ic_mass = particle.constant_index (it, "mass");
+        dens = *((enzo_float *)(particle.constant_value (it,ic_mass)));
+      } else {
+        dens = 1.0;
+        ia_m = particle.attribute_index(it, "mass");
```

Comment:
I find this very confusing, why is 'dens' used to refer to particle mass, but only in the case where mass is constant?

Reply:
I agree... but I think this is an idiom left over from Enzo where particle masses were stored as densities (mass / volume of a single cell)

Reply:
But also this routine takes the particle mass and deposits it onto the grid as a density, so not totally incorrect to have things named as dens

Reply:
I think this is something that needs to be cleared up... it's very confusing to have an attribute called 'mass' which is really density. Perhaps a way to fix it would be to have separate 'mass' and 'density' attributes, and this function could check for the separate cases. But this is something I can work on myself.

Reply:
Is this perhaps something that we can note in comments around the code in question, create an issue in the issue tracker, and then move forward?  @stefanarridge , I understand your concerns but this is a long-standing syntactic problem that is an Enzo-E issue more generally; it's not necessary to get it fixed in this PR, I don't think.